pkg/reconciler/intevents/reconciler.go

```diff
@@ -175,15 +189,101 @@ func (psb *PubSubBase) ReconcilePullSubscription(ctx context.Context, pubsubable
 		}
 	}
 
+	// TODO Compare the actual PullSubscription we found with the desired one and update as needed.
+
 	if err := propagatePullSubscriptionStatus(ps, status, cs); err != nil {
 		logging.FromContext(ctx).Desugar().Error("Failed to propagate PullSubscription status: %s", zap.Error(err))
 		return ps, pkgreconciler.NewEvent(corev1.EventTypeWarning, PullSubscriptionStatusPropagateFailedReason, "Failed to propagate PullSubscription status: %s", err.Error())
 	}
 
 	status.SinkURI = ps.Status.SinkURI
+
+	// The old and new PullSubscriptions can co-exist without any problems. So to bias in favor of
+	// double event delivery over dropped events, don't delete the old one until the new one is
+	// ready.
+	if ps.Status.IsReady() {
+		err = psb.deleteOldPubSubPullSubscriptionCO(ctx, pubsubable, ps)
+		if err != nil {
+			return ps, err
+		}
+	}
+
 	return ps, nil
 }
 
+func (psb *PubSubBase) deleteOldPubSubTopicCO(_ context.Context, pubsubable duck.PubSubable, t *inteventsv1alpha1.Topic) pkgreconciler.Event {
```

Comment:
nit: need the CO at the end? same for the pullSubscription deletion

Reply:
something that has been useful for me in this deprecation stuff has been to add TODOs saying remove after 0.x is cut, or something like that... not strong opinion, but might be easier for others to know what to remove

Reply:
Done.